SmartDeviceLinkTests/DevAPISpecs/SDLChoiceSetManagerSpec.m

```diff
@@ -440,105 +457,384 @@ - (void)sdl_addUniqueNamesToCells:(NSMutableSet<SDLChoiceCell *> *)choices;
             __block id<SDLChoiceSetDelegate> choiceDelegate = nil;
             __block id<SDLKeyboardDelegate> keyboardDelegate = nil;
             __block SDLInteractionMode testMode = SDLInteractionModeBoth;
             __block SDLPresentKeyboardOperation *pendingPresentOp = nil;
+            __block id strickMockOperationQueue = nil;
+            __block SDLChoiceCell *testSelectedCell = nil;
+            __block NSError *testError = nil;
+            NSUInteger testSelectedCellRow = 1;
 
             beforeEach(^{
                 keyboardDelegate = OCMProtocolMock(@protocol(SDLKeyboardDelegate));
                 choiceDelegate = OCMProtocolMock(@protocol(SDLChoiceSetDelegate));
                 testChoiceSet = [[SDLChoiceSet alloc] initWithTitle:testTitle delegate:choiceDelegate choices:@[testCell1, testCell2, testCell3]];
+                testFailedChoiceSet = [[SDLChoiceSet alloc] initWithTitle:testTitle delegate:choiceDelegate choices:@[testCell1, testCell2, testCell3, testCell4]];
+                testSelectedCell = testChoiceSet.choices[1];
+                testError = [NSError sdl_choiceSetManager_failedToCreateMenuItems];
 
                 pendingPresentOp = OCMClassMock([SDLPresentKeyboardOperation class]);
                 testManager.pendingPresentOperation = pendingPresentOp;
                 testManager.pendingPresentationSet = [[SDLChoiceSet alloc] init];
+
+                strickMockOperationQueue = OCMStrictClassMock([NSOperationQueue class]);
+                testManager.transactionQueue = strickMockOperationQueue;
             });
 
             context(@"searchable", ^{
-                beforeEach(^{
+                it(@"should notify the choice delegate when a choice item is selected", ^{
+                    OCMExpect([strickMockOperationQueue addOperation:[OCMArg checkWithBlock:^BOOL(id value) {
+                        SDLPreloadChoicesOperation *preloadChoicesOperation = (SDLPreloadChoicesOperation *)value;
+                        expect(testManager.pendingPresentationSet).to(equal(testChoiceSet));
+                        [preloadChoicesOperation finishOperation];
+                        return [value isKindOfClass:[SDLPreloadChoicesOperation class]];
+                    }]]);
+                    OCMExpect([strickMockOperationQueue addOperation:[OCMArg checkWithBlock:^BOOL(id value) {
+                        SDLPresentChoiceSetOperation *presentChoicesOperation = (SDLPresentChoiceSetOperation *)value;
+                        presentChoicesOperation.selectedCell = testSelectedCell;
+                        presentChoicesOperation.selectedTriggerSource = testMode;
+                        presentChoicesOperation.selectedCellRow = testSelectedCellRow;
+                        presentChoicesOperation.internalError = nil;
+                        presentChoicesOperation.completionBlock();
+                        return [value isKindOfClass:[SDLPresentChoiceSetOperation class]];
+                    }]]);
```

Comment:
Isn't the second expect here just overriding the first since it's on the same method? I'm not sure.

Reply:
The first `OCMExpect` is needed to trigger the `finishOperation` for the `SDLPreloadChoicesOperation` so the `SDLPresentChoiceSetOperation` can be created and sent. If you comment out either `OCMExpect`, the test fails.